src/rust/src/x509/ocsp_resp.rs

```diff
@@ -696,3 +744,101 @@ pub(crate) fn add_to_module(module: &pyo3::prelude::PyModule) -> pyo3::PyResult<
 
     Ok(())
 }
+
+#[pyo3::prelude::pyclass]
+struct OCSPResponseIterator {
+    contents: OwnedRawOCSPResponse,
+    cur_pos: usize,
```

Comment:
Rather than indices, can we use the iterator, as seen in CRLs? That'll avoid the `usize::MAX` stuff as well as the `O(n^2)` behavior.